storage/client/queries/queries.go

```diff
@@ -286,30 +286,67 @@ const (
+					entities.id = nodes.entity_id AND
+					nodes.roles LIKE '%validator%'
+				GROUP BY entities.address
+		)
 		SELECT
-				chain.entities.id AS entity_id,
-				chain.entities.address AS entity_address,
-				chain.nodes.id AS node_address,
-				chain.accounts.escrow_balance_active AS escrow,
-				COALESCE(chain.commissions.schedule, '{}'::JSONB) AS commissions_schedule,
-				EXISTS(SELECT NULL FROM chain.nodes WHERE chain.entities.id = chain.nodes.entity_id AND voting_power > 0) AS active,
-				EXISTS(SELECT NULL FROM chain.nodes WHERE chain.entities.id = chain.nodes.entity_id AND chain.nodes.roles like '%validator%') AS status,
-				chain.entities.meta AS meta
-			FROM chain.entities
-			JOIN chain.accounts ON chain.entities.address = chain.accounts.address
-			LEFT JOIN chain.commissions ON chain.entities.address = chain.commissions.address
-			JOIN chain.nodes ON chain.entities.id = chain.nodes.entity_id
-				AND chain.nodes.roles like '%validator%'
-				AND chain.nodes.voting_power = (
-					SELECT max(voting_power)
-					FROM chain.nodes
-					WHERE chain.entities.id = chain.nodes.entity_id
-						AND chain.nodes.roles like '%validator%'
-				)
-		WHERE ($1::text IS NULL OR chain.entities.address = $1::text)
+			chain.entities.id AS entity_id,
+			chain.entities.address AS entity_address,
+			chain.nodes.id AS node_address,
+			chain.accounts.escrow_balance_active AS active_balance,
+			chain.accounts.escrow_total_shares_active AS active_shares,
+			chain.accounts.escrow_balance_debonding AS debonding_balance,
+			chain.accounts.escrow_total_shares_debonding AS debonding_shares,
+			COALESCE (
+				ROUND(COALESCE(self_delegations.shares, 0) * chain.accounts.escrow_balance_active / NULLIF(chain.accounts.escrow_total_shares_active, 0))
+			, 0) AS self_delegation_balance,
+			COALESCE (
+				self_delegations.shares
+			, 0) AS self_delegation_shares,
+			COALESCE (
+				delegators_count.count
+			, 0) AS num_delegators,
+			active_nodes.voting_power,
+			COALESCE (
+				(SELECT SUM(voting_power)
+				FROM active_nodes)
+			, 0) AS total_voting_power,
+			COALESCE(chain.commissions.schedule, '{}'::JSONB) AS commissions_schedule,
+			chain.entities.start_date AS start_date,
+			ROW_NUMBER() OVER (ORDER BY escrow_balance_active DESC) AS rank,
+			EXISTS(SELECT NULL FROM chain.nodes WHERE chain.entities.id = chain.nodes.entity_id AND voting_power > 0) AS active,
+			EXISTS(SELECT NULL FROM chain.nodes WHERE chain.entities.id = chain.nodes.entity_id AND chain.nodes.roles like '%validator%') AS status,
+			chain.entities.meta AS meta
+		FROM chain.entities
+		JOIN chain.accounts ON chain.entities.address = chain.accounts.address
+		LEFT JOIN chain.commissions ON chain.entities.address = chain.commissions.address
+		LEFT JOIN self_delegations ON chain.entities.address = self_delegations.address
+		LEFT JOIN delegators_count ON chain.entities.address = delegators_count.address
+		JOIN active_nodes ON active_nodes.address = entities.address
```

Comment:
I believe this query currently returns all `chain.entities` even those that were never registered as validators (and don't intend to be).

I think this matches what oasiscan returns and I think it does actually make sense to return this. That way if some validator goes offline, you will still see it in the list of inactive validators.

I would just add a clarification to the API documentation for validator endpoints. Maybe reword it to:

> Returns a list of consensus validators (the list includes all registered entities even those without currently active validator node).